src/VisualStudio/Core/Def/Implementation/SymbolSearch/RoslynSymbolResult.cs

```diff
@@ -18,37 +19,65 @@
+    internal class RoslynSymbolResult : SymbolSearchResult, IResultInLocalFile, /*IResultWithDecoratedDefinition,*/ IResultWithClassifiedContext, IResultInNamedProject, IResultWithVSGuids, IResultWithKind, IResultInNamedCode
     {
-        private DefinitionItem definition;
-        private SourceReferenceItem reference;
-        private RoslynSymbolSource source;
-        private static ISymbolOriginDefinition LocalSymbolOrigin = null;
+        private DefinitionItem Definition { get; set; }
+        private SourceReferenceItem Reference { get; set; }
+        private RoslynSymbolSource Source { get; }
 
-        public override string Name { get; }
-        public override string Origin { get; }
-        public override ISymbolSource Owner { get; }
+        public override string Name => string.Empty;
+        public override string Origin => PredefinedSymbolOrigins.LocalCode;
+        public override ISymbolSource Owner => Source;
 
-        private RoslynSymbolResult(string name, RoslynSymbolSource owner)
+        private RoslynSymbolResult(RoslynSymbolSource source)
         {
-            this.Owner = owner;
-            this.Name = name;
-            this.Origin = PredefinedSymbolOrigins.LocalCode;
+            this.Source = source;
         }
 
         internal static async Task<RoslynSymbolResult> MakeAsync(RoslynSymbolSource symbolSource, DefinitionItem definition, DocumentSpan documentSpan, CancellationToken token)
         {
-            var result = new RoslynSymbolResult(definition.NameDisplayParts.FirstOrDefault().Text, symbolSource);
-            result.source = symbolSource;
-            result.definition = definition;
+            var result = new RoslynSymbolResult(symbolSource);
+            result.Definition = definition;
+            await MakeContent(result, documentSpan, token).ConfigureAwait(false);
+            return result;
+        }
+
+        internal static async Task<RoslynSymbolResult> MakeAsync(RoslynSymbolSource symbolSource, SourceReferenceItem reference, DocumentSpan documentSpan, CancellationToken token)
+        {
+            // TODO: Understand how Roslyn calls GetReferenceGroupsAsync
+            // it seems that at this time, we have full knowledge of definitions
+            // and their references
+
+            var result = new RoslynSymbolResult(symbolSource);
+            result.Reference = reference;
+            result.Definition = reference.Definition;
+            await MakeContent(result, documentSpan, token).ConfigureAwait(false);
+
+            return result;
+        }
 
+        private static async Task MakeContent(RoslynSymbolResult result, DocumentSpan documentSpan, CancellationToken token)
+        {
+            if (result.Reference != null)
+            {
+                result.Kind = result.Reference.SymbolUsageInfo.IsWrittenTo()
+                    ? "Write"
+                    : result.Reference.SymbolUsageInfo.IsReadFrom()
+                        ? "Read"
```

Comment:
constants?

what about Read/Write?  what about all the other sorts of SymbolUsageInfo?  do we just lose that?